ts/src/config.ts

```diff
@@ -61,25 +61,35 @@ export interface Config extends AuthenticationConfig {
   disableHeap?: boolean;
 }
 
-/**
- * Subtype of Config that declares some properties to be non-optional. This
- * helps avoid null checks in the profiler code.
- */
-export interface ProfilerConfig extends Config {
+// Interface for an initialized config.
+export interface ProfilerConfig extends AuthenticationConfig {
+  projectId?: string;
   logLevel: number;
-  serviceContext: {service?: string; version?: string;};
+  serviceContext: {service: string; version?: string;};
   instance: string;
   zone: string;
   disableTime: boolean;
   disableHeap: boolean;
+  timeIntervalMicros: number;
+  heapIntervalBytes: number;
+  heapMaxStackDepth: number;
+  backoffMillis: number;
```

Comment:
I don't think we should expose it in the config.

Reply:
Right now, users cannot set this value -- it would be overridden by the value in internalConfig.
Is there a better way to do this?

Reply:
Done.